src/app/shared/post-it-collections/collection-name/post-it-collection-name.component.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {PostItCollectionModel} from '../post-it-collection-model';
+
+@Component({
+  selector: 'post-it-collection-name',
+  templateUrl: './post-it-collection-name.component.html',
+  styleUrls: ['./post-it-collection-name.component.scss']
+})
+export class PostItCollectionNameComponent {
+
+  private readonly COLLECTION_NAME_COLLAPSED_HEIGHT = 42;
+
+  private readonly COLLECTION_NAME_MAX_HEIGHT = 500;
+
+  private readonly COLLECTION_PLACEHOLDER_NAME = 'Name';
+
+  private readonly COLLECTION_WHITESPACE_COLLAPSED = 'nowrap';
+
+  private readonly COLLECTION_WHITESPACE_EXPANDED = 'normal';
+
+  @Input()
+  public postIt: PostItCollectionModel;
+
+  @Input()
+  public selected: boolean;
+
+  @Output()
+  public changed = new EventEmitter();
+
+  public nameHeight(): string {
+    if (this.selected) {
+      return `${ this.COLLECTION_NAME_MAX_HEIGHT }px`;
+    } else {
+      return `${ this.COLLECTION_NAME_COLLAPSED_HEIGHT }px`;
+    }
+  }
+
+  public nameWrapping(): string {
+    if (this.selected) {
+      return this.COLLECTION_WHITESPACE_EXPANDED;
+    } else {
+      return this.COLLECTION_WHITESPACE_COLLAPSED;
+    }
+  }
+
+  public postItNamePlaceholder(collectionName: HTMLDivElement): string {
+    if (this.postIt.collection && this.postIt.collection.id) {
+      return '';
+    }
+
+    if (this.nameFocused(collectionName)) {
+      return '';
+    }
+
+    return this.COLLECTION_PLACEHOLDER_NAME;
+  }
+
+  public nameFocused(collectionName: HTMLDivElement): boolean {
+    return document.activeElement === collectionName;
+  }
+
+  public onNameChanged(newCollectionName: HTMLDivElement) {
+    this.postIt.collection.name = newCollectionName.textContent;
```

Comment:
It is not a good idea to modify internals of the object passed in as the component input. It may introduce some unwanted side effects.

What about keeping the changed name in some other private property, then compare it with the original value on blur and send it in `changed` event only if the value has really been changed?

Reply:
Can't think of a case, where this would cause issues.
But ok. I'll change it.

It will require changing post-it-collections to NgRx.. 
since this way of expecting the store to update the values doesn't work with our services.